### PR TITLE
fix(auth-server): subplat email images are at 100% width using the Mail app

### PR DESCRIPTION
**Pull request by xlisachan**
## Because

- per [MJML documentation](https://documentation.mjml.io/#mj-image), a `<mj-image>` tag displays a responsive center-aligned image in an email and, if no width is provided, the image will use the parent column width. When the `<mj-image>` tag is translated into HTML, the image is nested within a `<td>` tag, which utilizes the width attribute set in the `<mj-image>` tag. While the width of the image can be set in a `css-class`, it only sets the image width and not the width of the `<td>` it is nested in. Therefore, it is best practice to also include and set the width attribute when using a `<mj-image>` tag.

Note: The header image within the FxA email templates did not appear at 100% width in the Mail app, as it was written in HTML within a `<mj-raw>` tag and no `<mj-image>` tags were utilized.

Screen Shot 2022-02-03 at 9 14 53 AM

## This pull request

- Converts the following images in FxA email template to MJML format, sets the width within their `mj-image` tags, and removes their widths in their `css-class`:
  - Header Firefox logo
  - Header Sync image
- Sets the width attribute within all of the following `mj-image` tags  and removes its width in its `css-class`:
  - Header Firefox logo of all subplat email templates
  - Footer Mozilla logo of all subplat email templates
  - App Store badge
  - Google Play badge
  - Product icons within the `icon` partial
  - Device images used in `cadReminderFirst`, `cadReminderSecond`, `postVerify`
- Adds FxA images (header logo and sync devices image) to `emails/partials/images.mjml` which localizes the image alt text

## Issue that this pull request solves

Closes: #11833 

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [ ] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

### CODE REVIEW

**Comment by LZoog** [comment]
Thanks as usual for jumping on this bug!

> Note: The header image within the FxA email templates did not appear at 100% width in the Mail app, as it was written in HTML within a `<mj-raw>` tag and no `<mj-image>` tags were utilized.

Do you know why it was done that way? It feels like maybe that was a mistake since that layout file was one of the first files we wrote in mjml. If you're not sure why (I don't at first glance, but I could be missing something) I think we should change that for consistency's sake - I only see those two instances where we're using the pure `img`, and they're both in that layout file.

Also, should we take all of the CSS class `width`s out if they're not actually doing anything? That way we don't have to set the width in two places.

**Comment by LZoog** [comment]
One other thing - in checking SubPlat emails off of `main` in Storybook, the rendered HTML is this...

image

And then in Gmail, I'm seeing the same HTML output.

image

I'm cool with seeing if this fixes it, but I'm not sure how the HTML output changes 🤔

**Comment by xlisachan** [comment]
> Thanks as usual for jumping on this bug!
> 
> > Note: The header image within the FxA email templates did not appear at 100% width in the Mail app, as it was written in HTML within a `<mj-raw>` tag and no `<mj-image>` tags were utilized.
> 
> Do you know why it was done that way? It feels like maybe that was a mistake since that layout file was one of the first files we wrote in mjml. If there's not a compelling reason to leave it (I don't see it at first glance, but I could be missing something) I think we should change those to `mj-image` for consistency's sake - I only see those two instances where we're using the pure `img`, and they're both in that layout file. Either way, I can make note in my documentation ticket this sprint when to use `<mj-raw>` and `` and explicitly setting the width there.
> 

Unfortunately that was before my time, my guess would be because of the conditionals. You know I am always down for consistency 😎 I'll let you know if I run into any issues with the `preHeader` portion.

> Also, should we take all of the CSS class `width`s out if they're not actually doing anything? That way we don't have to set the width in two places.

I'm aligned with taking the widths out of `css-class` since the image width takes on the attribute in `<mj-image>`. I'll push this change as well.

**Comment by xlisachan** [comment]
> One other thing - in checking SubPlat emails off of `main` in Storybook, the rendered HTML is this...
> 
> And then in Gmail, I'm seeing the same HTML output.
> 
> I'm cool with seeing if this fixes it, but I'm not sure how the HTML output changes 🤔
> 
> EDIT: After posting this I realized `550` was entirely too large for that to be correct. It's got to be coming from this `td` parent even further up the DOM. 
> 
> I see the same thing on the VPN image. image
> 
> ... This makes me wonder if Mail isn't respecting that `td` width way further up, and is just displaying the full `550` width? Where the heck is 550 even coming from?

Lol that was my rant last night. So when the width is not set in the `<mj-image>` tag, it automatically set to the parent, I am not sure why that is `550`, but `550` is even used in the MJML sandbox so I am guessing that is the default width of an email.

Once the fix is applied the width of the `<td>` that the image is nested in is updated, so it should be fine.

Screen Shot 2022-02-03 at 2 30 25 PM

File: packages/fxa-auth-server/lib/senders/emails/layouts/fxa/index.mjml
Comment (xlisachan) [inline packages/fxa-auth-server/lib/senders/emails/layouts/fxa/index.mjml:31]: I need another set of eyes on the preHeader section.

File: packages/fxa-auth-server/lib/senders/emails/layouts/subscription/index.mjml
Comment (xlisachan) [inline packages/fxa-auth-server/lib/senders/emails/layouts/subscription/index.mjml:35]: I need another set of eyes on the preHeader section too.

File: packages/fxa-auth-server/lib/senders/emails/layouts/subscription/index.mjml
Comment (xlisachan) [inline packages/fxa-auth-server/lib/senders/emails/layouts/subscription/index.mjml:41]: Default value for align is center.

**Review by xlisachan — COMMENTED** [review]
I'm not sure if I've ever seen a `preHeader` in any email...

File: packages/fxa-auth-server/lib/senders/emails/partials/icon/index.mjml
Comment (pdehaan) [inline packages/fxa-auth-server/lib/senders/emails/partials/icon/index.mjml:6]: What language is this? EJS?
I tried reading the mjml docs but didn't see anything obvious re:scripting languages.

Curious if there's a way we could make a [slightly gross] helper function that could determine the image dimensions from the image itself instead of hardcoding values like "58px".
I've used http://npm.im/image-size in the past with some success (albeit mostly just linting a series of logos to verify that they were all roughly 150&times;150px).

Comment (xlisachan) [reply packages/fxa-auth-server/lib/senders/emails/partials/icon/index.mjml:6]: > What language is this? EJS? I tried reading the mjml docs but didn't see anything obvious re:scripting languages.
> 
Yep, EJS. 

> Curious if there's a way we could make a [slightly gross] helper function that could determine the image dimensions from the image itself instead of hardcoding values like "58px". I've used http://npm.im/image-size in the past with some success (albeit mostly just linting a series of logos to verify that they were all roughly 150×150px).
>
Hmm, I've only used `image-size` to resize an uploaded image before it's stored and saved as a hero image/thumbnail when I was on AMO. We'll file a follow-up maintenance item for this if we can utilize it here. Thanks for the suggestion!

File: packages/fxa-auth-server/lib/senders/emails/layouts/fxa/index.mjml
Comment (LZoog) [inline packages/fxa-auth-server/lib/senders/emails/layouts/fxa/index.mjml:44]: Do we need anything else to ensure `title` is localized or does what you have set in the `images` partial cover us here? Fine to get up in another PR if so, the fix LGTM so we can get this into the dot release Ben is doing. 😄

**Review by LZoog — APPROVED** [review]
:shipit:

File: packages/fxa-auth-server/lib/senders/emails/layouts/fxa/index.mjml
Comment (xlisachan) [reply packages/fxa-auth-server/lib/senders/emails/layouts/fxa/index.mjml:44]: The `images` partial/en.ftl only cover the image alt text. We might want to revisit this at a later time. We would have to edit all our en.ftl's to include `title`, but this might mean we would need new ftl IDs...

**Pull request merged**
